modules/database/test/std/rec/printfTest.c

```diff
@@ -0,0 +1,568 @@
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_LONG, 0xfafa);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_X_format(void){
+    const char format_string[] = "Format test string %X";
+    const char result_string[] = "Format test string BA";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 0x00ba);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_e_format(void){
+    const char format_string[] = "Format test string %e";
+    const char result_string[] = "Format test string -1.400000e+01";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, -14);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_E_format(void){
+    const char format_string[] = "Format test string %E";
+    const char result_string[] = "Format test string 1.992000E+03";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_SHORT, 1992);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_f_format(void){
+    const char format_string[] = "Format test string %f";
+    const char result_string[] = "Format test string -0.062000";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_DOUBLE, -0.062);
+
+    /* verify that string is formatted as expected */
+    testdbGetFieldEqual("test_printf_rec.VAL", DBF_STRING, result_string);
+
+    // number of tests = 3
+}
+
+static void test_F_format(void){
+    const char format_string[] = "Format test string %F";
+    const char result_string[] = "Format test string 6729982.999000";
+
+    /* set format string */
+    testdbPutFieldOk("test_printf_rec.FMT", DBF_STRING, format_string);
+
+    /* set value on inp0 */
+    testdbPutFieldOk("test_printf_inp0_rec.VAL", DBF_DOUBLE, 6729982.999);
+
+    /* verify that string is formatted as expected */
+    #ifdef _WIN32
```

Comment:
Is this possibly a gcc rather than Win2019/mingw specific issue? @ralphlange mentioned this platform used gcc 8.* and I see the ubuntu tests use gcc 9 and pass, so maybe using
```
#if defined(__GNUC__)  &&  __GNUC__ < 9
```
would be sufficient

Reply:
I see the Appveyor Win2019 mingw build uses gcc 12.2.0 so if this issues is gcc version specific then that build should pass

Reply:
I suspect this may not have anything to do with the %F specifier and everything to do with the fact that the number I am trying to convert does not fit into a float without truncation. Testing a bit to see if I can get rid of this ugly todoTest.

Reply:
I was wrong. A float with fewer digits does not fare any better.

Reply:
Appveyor shows that several windows based platforms fails, not only on %F but also on %lx. I have skipped these two tests on all windows platforms for now and updated issue #368 accordingly. Resolving this conversation in favor of the issue for future discussion.